### ISSUE TITLE
Nyxt window crashes after loading default history and changing to a web buffer

### ISSUE THREAD

**Issue by fade** [atlas-engineer/nyxt#1840]
**Describe the bug**
When starting nyxt built from either 2.2.2 tag, or the HEAD of the master branch, it crashes after selecting the default history file and switching to a web buffer. 
**Precise recipe to reproduce the issue**
Start Nyxt, select default history, switch to web buffer. <crash>
This still happens when nyxt is started without any configuration.
prior to switching to the web buffer, the minibuffer shows: 
Warning: Errorr on GTK thread: The value :INVALID-CODE-OBJECT-AT-PC is not of type FUNCTION
**Information**
- OS name+version: XeroLinux (Arch, rolling) 
- Linux kernel: 5.14.14-arch1-1 #1 SMP PREEMPT Wed, 20 Oct 2021 21:35:18 +0000 x86_64
- Graphics card and driver: NVIDIA Quadro K1000M/Intel 3rd Gen Core GC running noveau/intel on XORG/X11
- Desktop environment / Window manager name+version: StumpWM 20.11-91-gf867f9e Compiled On Fri Oct 29 2021 22:26:29
- How you installed Nyxt (Guix pack, package manager, build from source): Built Nyxt from source, both 2.2.2 tag and HEAD of master, although I see that as of right now, these things are the same commit.
- Information from `show-system-information`:

Nyxt version: 2.1.1-713-g26a49cb0
Renderer version: GI-GTK
Operating system kernel: Linux 5.14.14-arch1-1
Lisp implementation: SBCL 2.1.9 (Dynamic space size: 12884901888)
Features: (:WEBKIT2 :WEBKIT2-2.34 :WEBKIT2-PASTE-PLAINTEXT :WEBKIT2-TRACKING
 :WEBKIT2-MUTE :WEBKIT2-EMOJI :WEBKIT2-MEDIA :WEBKIT2-SANDBOXING :GTK-3-22
 :GTK-3-20 :GTK-3-18 :GTK-3-16 :GTK-3-14 :GTK-3-12 :GTK-3-10 :GTK-3-8 :GTK-3-6
 :GTK-3-4 :GTK :GDK-3-22 :GDK-3-20 :GDK-3-18 :GDK-3-16 :GDK-3-14 :GDK-3-12
 :GDK-3-10 :GDK-3-8 :GDK-3-6 :GDK-3-4 :CAIRO-1-10 :CAIRO-1-12 :GDK-PIXBUF
 :GLIB-2-30 :GLIB-2-32 :GLIB-2-34 :GLIB-2-36 :GLIB-2-38 :GLIB-2-40 :GLIB-2-42
 :GLIB-2-44 :GLIB-2-46 :GLIB-2-48 :GLIB-2-50 :GLIB-2-52 :GLIB-2-54 :GLIB-2-56
 :GLIB-2-58 :GLIB :NYXT-2 :FSET-EXT-STRINGS :CUSTOM-HASH-TABLE-NATIVE :SWANK
 :PLUMP-UTF-32 :GLOBAL-VARS :DECLARE-TYPES :PARENSCRIPT
 :SBCL+SAFE-STANDARD-READTABLE :NAMED-READTABLES :LPARALLEL :21BIT-CHARS
 :CHUNGA :CLOSER-MOP :CL-PPCRE-UNICODE :FLEXI-STREAMS :CL-UNICODE :CL-PPCRE
 :CL-JSON-DOUBLE-FLOAT-IS-SUBSUMED :CL-JSON-SINGLE-FLOAT-IS-SUBSUMED
 :BORDEAUX-THREADS :LPARALLEL.WITH-CLTL2 :LPARALLEL.WITH-CAS
 :LPARALLEL.WITH-STEALING-SCHEDULER :SPLIT-SEQUENCE CHIPZ-SYSTEM:GRAY-STREAMS
 CFFI-FEATURES:FLAT-NAMESPACE CFFI-FEATURES:X86-64 CFFI-FEATURES:UNIX :CFFI
 CFFI-SYS::FLAT-NAMESPACE ALEXANDRIA::SEQUENCE-EMPTYP :FAST-IO-SV :FAST-IO
 :SBCL-USES-SB-ROTATE-BYTE :CL-JSON-CLOS :CL-JSON :THREAD-SUPPORT :ASDF3.3
 :ASDF3.2 :ASDF3.1 :ASDF3 :ASDF2 :ASDF :OS-UNIX :NON-BASE-CHARS-EXIST-P
 :ASDF-UNICODE :X86-64 :GENCGC :64-BIT :ANSI-CL :COMMON-LISP :ELF
 :IEEE-FLOATING-POINT :LINUX :LITTLE-ENDIAN :PACKAGE-LOCAL-NICKNAMES
 :SB-CORE-COMPRESSION :SB-LDB :SB-PACKAGE-LOCKS :SB-THREAD :SB-UNICODE :SBCL
 :UNIX)

ASDF version: 3.3.1
Critical dependencies: (/home/fade/SourceCode/lisp/nyxt/_build/cl-cffi-gtk/gtk/cl-cffi-gtk.asd
 /home/fade/SourceCode/lisp/nyxt/_build/cl-gobject-introspection/cl-gobject-introspection.asd
 /home/fade/SourceCode/lisp/nyxt/_build/cl-webkit/webkit2/cl-webkit2.asd)

  - Nyxt version (from the `nyxt-version` command or `nyxt --version` from a shell):  Nyxt version 2.1.1-739-g43862f67
  - Lisp implementation/version (if built from source): sbcl 2.1.10
  - Kernel name+version: Linux ghost 5.14.14-arch1-1 #1 SMP PREEMPT Wed, 20 Oct 2021 21:35:18 +0000 x86_64 GNU/Linux
  - WebKitGTK+ | QtWebEngine version: webkit2gtk 2.34.1-1

**Output when started from a shell**
<INFO> [10:30:37] Listening to socket "/run/user/1000/nyxt/nyxt.socket".
Nyxt version 2.1.1-713-g26a49cb0
<INFO> [10:30:37] Loading Lisp file "/home/fade/.config/nyxt/auto-config.lisp".
<INFO> [10:30:43] Loading history of 154 URLs from "/home/fade/.local/share/nyxt/history/default.lisp".
<INFO> [10:30:43] Loading history of 154 URLs from "/home/fade/.local/share/nyxt/history/default.lisp".
<INFO> [10:30:43] Restoring 15 buffers from history.

** (nyxt:92378): ERROR **: 10:30:44.745: GApplication is required for xdg-desktop-portal access in the WebKit sandbox.
<WARN> [10:30:44] Warning: Error on GTK thread: The value
                       NIL
                     is not of type
                       SB-KERNEL:CODE-COMPONENT

** (process:92378): ERROR (recursed) **: GApplication is required for xdg-desktop-portal access in the WebKit sandbox.fatal error encountered in SBCL pid 92378 tid 92396:
SIGABRT received.

   0: fp=0x7f97141f68f0 pc=0x7f9714070d22 Foreign function gsignal
[1]    92378 exit 1     nyxt

**Comment by fade** [comment]
setting and exporting WEBKIT_FORCE_SANDBOX=0 and rebuilding the application with the gtk-application (as opposed to the default gi-gtk) renderer allows the application to come up and function normally. With thanks to @waleee on the #nyxt libera irc channel.

**Comment by Ambrevar** [comment]
Thanks for reporting, but something else is fishy in your reports, so you might want to double-check the Git log of the repo and what you've compiled.
Can you share the commands you use to build Nyxt?

- `2.2` is _not_ `master`, in fact 2.2 lives on a different branch.
- You reported version 2.1.1-713-g26a49cb0, which is 3 weeks old.

I believe that checking out to 2.2 should fix your issue.  Let me know if it works without setting `WEBKIT_FORCE_SANDBOX=0` then.

**Comment by OlMon** [comment]
Hej Hej,

I have the same problem/error. I tried both aur packages (nyxt and nyxt-browser-git)

nyxt-browser-git: 2.1.1-740-g37d0bae9
nyxt: 2.2.2

Setting `WEBKIT_FORCE_SANDBOX=0` let me use nyxt normally.

If you need any more information please let me know.

Cheers!

**Comment by sivark** [comment]
In case it's helpful, I'm observing the same behavior as @OlMon (crash, that goes away with the ENV variable tweak), with the same setup (AUR package `nyxt`). Here's my system information:
```
System information

Nyxt version: 2.2.2
Renderer version: GI-GTK
Operating system kernel: Linux 5.14.14-arch1-1
Lisp implementation: SBCL 2.1.9 (Dynamic space size: 1073741824)
Features: (:WEBKIT2 :WEBKIT2-2.34 :WEBKIT2-PASTE-PLAINTEXT :WEBKIT2-TRACKING
 :WEBKIT2-MUTE :WEBKIT2-EMOJI :WEBKIT2-MEDIA :WEBKIT2-SANDBOXING :GTK-3-22
 :GTK-3-20 :GTK-3-18 :GTK-3-16 :GTK-3-14 :GTK-3-12 :GTK-3-10 :GTK-3-8 :GTK-3-6
 :GTK-3-4 :GTK :GDK-3-22 :GDK-3-20 :GDK-3-18 :GDK-3-16 :GDK-3-14 :GDK-3-12
 :GDK-3-10 :GDK-3-8 :GDK-3-6 :GDK-3-4 :CAIRO-1-10 :CAIRO-1-12 :GDK-PIXBUF
 :GLIB-2-30 :GLIB-2-32 :GLIB-2-34 :GLIB-2-36 :GLIB-2-38 :GLIB-2-40 :GLIB-2-42
 :GLIB-2-44 :GLIB-2-46 :GLIB-2-48 :GLIB-2-50 :GLIB-2-52 :GLIB-2-54 :GLIB-2-56
 :GLIB-2-58 :GLIB :NYXT-2 :FSET-EXT-STRINGS :CUSTOM-HASH-TABLE-NATIVE :SWANK
 :PLUMP-UTF-32 :GLOBAL-VARS :DECLARE-TYPES :PARENSCRIPT
 :SBCL+SAFE-STANDARD-READTABLE :NAMED-READTABLES :LPARALLEL :21BIT-CHARS
 :CHUNGA :CLOSER-MOP :CL-PPCRE-UNICODE :FLEXI-STREAMS :CL-UNICODE :CL-PPCRE
 :CL-JSON-DOUBLE-FLOAT-IS-SUBSUMED :CL-JSON-SINGLE-FLOAT-IS-SUBSUMED
 :BORDEAUX-THREADS :LPARALLEL.WITH-CLTL2 :LPARALLEL.WITH-CAS
 :LPARALLEL.WITH-STEALING-SCHEDULER :SPLIT-SEQUENCE CHIPZ-SYSTEM:GRAY-STREAMS
 CFFI-FEATURES:FLAT-NAMESPACE CFFI-FEATURES:X86-64 CFFI-FEATURES:UNIX :CFFI
 CFFI-SYS::FLAT-NAMESPACE ALEXANDRIA::SEQUENCE-EMPTYP :FAST-IO-SV :FAST-IO
 :SBCL-USES-SB-ROTATE-BYTE :CL-JSON-CLOS :CL-JSON :THREAD-SUPPORT :ASDF3.3
 :ASDF3.2 :ASDF3.1 :ASDF3 :ASDF2 :ASDF :OS-UNIX :NON-BASE-CHARS-EXIST-P
 :ASDF-UNICODE :X86-64 :GENCGC :64-BIT :ANSI-CL :COMMON-LISP :ELF
 :IEEE-FLOATING-POINT :LINUX :LITTLE-ENDIAN :PACKAGE-LOCAL-NICKNAMES
 :SB-CORE-COMPRESSION :SB-LDB :SB-PACKAGE-LOCKS :SB-THREAD :SB-UNICODE :SBCL
 :UNIX)

ASDF version: 3.3.1
ASDF registries: (NYXT-SOURCE-REGISTRY ENVIRONMENT-SOURCE-REGISTRY)
Critical dependencies: (/tmp/nyxt-7425437938866822998/nyxt/src/nyxt/_build/cl-cffi-gtk/gtk/cl-cffi-gtk.asd
 /tmp/nyxt-7425437938866822998/nyxt/src/nyxt/_build/cl-gobject-introspection/cl-gobject-introspection.asd
 /tmp/nyxt-7425437938866822998/nyxt/src/nyxt/_build/cl-webkit/webkit2/cl-webkit2.asd)
```

LMK if I can be of assistance.

**Comment by Ambrevar** [comment]
I can't reproduce on 2.2.2.

Does the crash happen on specific web pages?
If so, try to load a page that does not contain videos like https://nyxt.atlas.engineer/.

@aartaka Could this be related to https://github.com/atlas-engineer/nyxt/issues/1799 ?

**Comment by Ambrevar** [comment]
@OlMon and @sivark use Arch, so I assume they are on WebKitGTK 2.34, which may what triggers the crash.  This may be a WebKitGTK bug then.

**Comment by OlMon** [comment]
@Ambrevar  It happens on any page, or at least on every I tried (more than a few). I assume it is not directly because of a nyxt update. I updated some packages (don't ask which one), but nyxt was not one of them. After that nyxt did crashed as described above.

**Comment by Ambrevar** [comment]
Can you reproduce on Nyxt 2.2.0?

**Comment by OlMon** [comment]
On 2.2.0 it is working, so maybe my statement above was not correct about the update. I am not sure anymore.

**Comment by aartaka** [comment]
> @aartaka Could this be related to #1799 ?

Yes, and to #1781.

**Comment by khinsen** [comment]
I see the same issue with 2.2.2 compiled under a freshly updated Guix. Went back to 2.2.0, problem solved.

**Comment by Ambrevar** [comment]
@aartaka I suggest reverting https://github.com/atlas-engineer/nyxt/pull/1505 for the `2-series` branch.

**Comment by jmercouris** [comment]
I concur, let's revert that change for the `2-series` branch and leave it on master for now. Master is experimental, after all.

**Comment by aartaka** [comment]
> I concur, let's revert that change for the `2-series` branch and leave it on master for now. Master is experimental, after all.

But that won't solve the problem -- `.config` is erroring too. See my comments in #1505.

**Comment by Ambrevar** [comment]
Looks like I can reproduce this issue on Guix with the recent WebKitGTK 2.34.1 update.
So it may be an upstream issue after all, or we were using the API wrong before :p

**Comment by Ambrevar** [comment]
All WebKitGTK 2.34-related hangs / crashes should be fixed on master since c2f1e61.

We had to stop enforcing the sandboxing though. We are tackling this in a separate issue, see #1781.
